### ISSUE TITLE
Analysis and synthesis

### ISSUE THREAD

**Issue by pedrottimark** [gaearon/ama#20]
What is the balance between analysis (taking things apart) and synthesis (putting things together) in your thinking process?
How do you express those gifts or skills outside of your open source work?

**Comment by gaearon** [comment]
If I want to understand how a library works, I put console.logs everywhere. If I don't understand something I don't worry and forget about it. I still don't understand a lot of parts in React, for example.

To "put things together" I like to try to reimplement them. This let me see the obstacles and complications, and then I can come back to the code I'm trying to understand with a new understanding of the challenges and constraints its authors had in their minds.

I'm trying to see what people in relatively disconnected parts of the ecosystem do in case I can figure out how to connect their creations. I'm bad at inventing new stuff, but I like to notice when some projects or ideas can be connected together to create something greater than sum of its parts. For example, I created React Hot Loader when I noticed Webpack's Hot Module Replacement and React go together really well.
